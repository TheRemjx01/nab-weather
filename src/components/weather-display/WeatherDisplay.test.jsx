import React from 'react';
import {shallow} from 'enzyme';

import {
 generateTitle,
 generateLocationForecastUrl,
 WeatherDisplayData,
 WeatherDisplay,
 WeatherError,
 WeatherLoading,
} from './WeatherDisplay';

describe('generateTitle', () => {
 it('should be able to generate title', () => {
  expect(generateTitle({title: 'London'})).toBe('London Weather')
 })
})

describe('generateLocationForecastUrl', () => {
 it('should be able to generate correct url', () => {
  expect(generateLocationForecastUrl({locationId: 1005})).toBe('/location/1005')
 })
})

describe('<WeatherLoading />', () => {
 it('should be able to render', () => {
  const Wrapper = shallow(<WeatherLoading />);
  expect(Wrapper).toMatchSnapshot();
 })
})

describe('<WeatherError />', () => {
 it('should be able to render', () => {
  const Wrapper = shallow(<WeatherError error='mock error'/>);
  expect(Wrapper).toMatchSnapshot();
 })
})

describe('<WeatherDisplay />', () => {
 it('should be able to render', () => {
  const Wrapper = shallow(<WeatherDisplay locationId={1105} locationLabel={'London'} />);
  expect(Wrapper).toMatchSnapshot();
 })
})

describe('<WeatherDisplayData />', () => {
 it('should be able to render when no data', () => {
  const Wrapper = shallow(<WeatherDisplayData />);
  expect(Wrapper).toMatchSnapshot();
 })

 it('should be able to render when has data', () => {
  const weatherData = [{
   id: 1, min_temp: 25, max_temp: 28, applicable_date: '2020-08-11', weather_state_abbr: 'sn',
  }]
  const Wrapper = shallow(<WeatherDisplayData data={weatherData}/>);
  expect(Wrapper).toMatchSnapshot();
 })
})

