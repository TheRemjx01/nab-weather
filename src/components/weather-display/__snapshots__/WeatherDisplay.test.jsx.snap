// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<Col
  key="1"
>
  <WeatherItem
    maxTemp={2}
    minTemp={1}
    weatherIcon="sn"
    weekDay="Tuesday"
  />
</Col>
`;

exports[`<WeatherDisplay /> should be able to render 1`] = `
<div>
  <h2
    className="locationLabel"
  >
    London
  </h2>
  <WithFetch
    Component={<WeatherDisplayData />}
    ErrorComponent={<WeatherError />}
    LoadingComponent={<WeatherLoading />}
    url="/api/location/1105"
  />
</div>
`;

exports[`<WeatherDisplayData /> should be able to render when has data 1`] = `
<Fragment>
  <Title
    text="undefined Weather"
  />
  <div
    className="weatherDisplay"
  >
    <Row />
  </div>
</Fragment>
`;

exports[`<WeatherDisplayData /> should be able to render when no data 1`] = `""`;

exports[`<WeatherError /> should be able to render 1`] = `"mock error"`;

exports[`<WeatherLoading /> should be able to render 1`] = `
<div
  className="weatherDisplay loading"
>
  <Spin
    size="large"
    spinning={true}
    wrapperClassName=""
  />
</div>
`;
